units/en/unit2/smolagents/code_agents.mdx

```diff
@@ -0,0 +1,378 @@
+<CourseFloatingBanner chapter={2}
+  classNames="absolute z-10 right-0 top-0"
+  notebooks={[
+    {label: "Google Colab", value: "https://colab.research.google.com/github/huggingface/agents-course/blob/main/notebooks/unit2/smolagents/code_agents.ipynb"},
+]} />
+
+# Building Agents That Use Code
+
+Code agents are the default agent type in `smolagents`. They generate Python tool calls directly to perform actions, making them both efficient and accurate. 
+
+Their streamlined approach reduces the number of required actions, simplifies complex operations, and enables reuse of existing code functions. `smolagents` provides a lightweight framework for building code agents, implemented in approximately 1,000 lines of code.
+
+![Code vs JSON Actions](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/code_vs_json_actions.png)
+Graphic from the paper [Executable Code Actions Elicit Better LLM Agents](https://huggingface.co/papers/2402.01030)
+
+<Tip>
+If you want to learn more about why code agents are effective, check out <a href="https://huggingface.co/docs/smolagents/en/conceptual_guides/intro_agents#code-agents" target="_blank">this guide</a> from the smolagents documentation. 
+</Tip>
+
+## Why Code Agents?
+
+In a multi-step agent process, the LLM writes and executes actions, typically involving external tool calls. Traditional approaches use a JSON format to specify tool names and arguments as strings, **which the system must parse to determine which tool to execute**.
+
+However, research shows that **tool-calling LLMs work more effectively with code directly**. This is a core principle of `smolagents`, as shown in the diagram above from [Executable Code Actions Elicit Better LLM Agents](https://huggingface.co/papers/2402.01030).
+
+Writing actions in code rather than JSON offers several key advantages:
+
+* **Composability**: Easily combine and reuse actions
+* **Object Management**: Work directly with complex structures like images
+* **Generality**: Express any computationally possible task
+* **Natural for LLMs**: High-quality code is already present in LLM training data
+
+## How Does a Code Agent Work?
+
+![From https://huggingface.co/docs/smolagents/conceptual_guides/react](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/smolagents/codeagent_docs.png)
+
+The diagram above illustrates how `CodeAgent.run()` operates, following the [ReAct framework](https://huggingface.co/papers/2210.03629), which is currently the preferred approach for building multi-step agents. We already introduced the `MultiStepAgent` in the previous section, which serves as the core building block for `smolagents`. The `CodeAgent` is a special kind of `MultiStepAgent`, as we will see in this example underneath.
```

Comment:
```suggestion
The diagram above illustrates how `CodeAgent.run()` operates, following the ReAct framework we mentioned in Unit 1. The main abstraction for agents in `smolagents` is a `MultiStepAgent`, which serves as the core building block. `CodeAgent` is a special kind of `MultiStepAgent`, as we will see in an example below.
```

I'd keep this light in paper references. Also, still not sure the `MultiStepAgent` reference is relevant until we show the code and/or other types of agents.